core/store/tx_manager_test.go

```diff
@@ -436,6 +436,100 @@ func TestTxManager_BumpGasUntilSafe_atGasBumpThreshold(t *testing.T) {
 	ethMock.EventuallyAllCalled(t)
 }
 
+func TestTxManager_BumpGasUntilSafe_atGasBumpThreshold_bumpsGasMoreInCaseOfUnderpricedTransaction(t *testing.T) {
+	t.Parallel()
+
+	app, cleanup := cltest.NewApplicationWithKey(t)
+	defer cleanup()
+
+	store := app.Store
+	config := store.Config
+
+	txm := store.TxManager
+	from := cltest.GetAccountAddress(t, store)
+	sentAt := uint64(23456)
+	gasThreshold := sentAt + config.EthGasBumpThreshold()
+	ethMock := app.EthMock
+	ethMock.Register("eth_getTransactionCount", "0x0")
+	ethMock.Register("eth_chainId", config.ChainID())
+	require.NoError(t, app.Store.ORM.CreateHead(cltest.Head(gasThreshold)))
+	require.NoError(t, app.StartAndConnect())
+
+	tx := cltest.CreateTxWithNonceAndGasPrice(t, store, from, sentAt, 0, 48000000000)
+	require.Greater(t, len(tx.Attempts), 0)
+
+	ethMock.Register("eth_getTransactionReceipt", eth.TxReceipt{})
+
+	// Simulate two bumps that receive `replacement transaction underpriced`
+	// and the third and final one successful
+	ethMock.RegisterError("eth_sendRawTransaction", "replacement transaction underpriced")
+	ethMock.RegisterError("eth_sendRawTransaction", "replacement transaction underpriced")
+	ethMock.Register("eth_sendRawTransaction", cltest.NewHash())
+
+	receipt, state, err := txm.BumpGasUntilSafe(tx.Attempts[0].Hash)
+	assert.NoError(t, err)
+	assert.Nil(t, receipt)
+	assert.Equal(t, strpkg.Unconfirmed, state)
+
+	tx, err = store.FindTx(tx.ID)
+	require.NoError(t, err)
+	assert.Len(t, tx.Attempts, 2)
+
+	latestAttempt := tx.Attempts[1]
+	gasPrice, err := latestAttempt.GasPrice.Value()
+	require.NoError(t, err)
+
+	// Initial gas price is 48Gwei
+	// Bump to 53Gwei and fail
+	// Bump to 58.3Gwei and fail
+	// Bump to 64.13Gwei and pass
+	assert.Equal(t, "64130000000", gasPrice)
+
+	ethMock.EventuallyAllCalled(t)
+}
+
+func TestTxManager_BumpGasUntilSafe_atGasBumpThreshold_returnsErrorIfMaxGasPriceIsReached(t *testing.T) {
+	t.Parallel()
+
+	app, cleanup := cltest.NewApplicationWithKey(t)
+	defer cleanup()
+
+	store := app.Store
+	config := store.Config
+
+	txm := store.TxManager
+	from := cltest.GetAccountAddress(t, store)
+	sentAt := uint64(23456)
+	gasThreshold := sentAt + config.EthGasBumpThreshold()
+	ethMock := app.EthMock
+	ethMock.Register("eth_getTransactionCount", "0x0")
+	ethMock.Register("eth_chainId", config.ChainID())
+	require.NoError(t, app.Store.ORM.CreateHead(cltest.Head(gasThreshold)))
+	require.NoError(t, app.StartAndConnect())
+
+	tx := cltest.CreateTxWithNonceAndGasPrice(t, store, from, sentAt, 0, 499000000000)
+	store.SaveTx(tx)
+	require.Greater(t, len(tx.Attempts), 0)
+
+	ethMock.Register("eth_getTransactionReceipt", eth.TxReceipt{})
+
+	receipt, state, err := txm.BumpGasUntilSafe(tx.Attempts[0].Hash)
+	assert.EqualError(t, err, "bumped gas price of 548900000000 would exceed maximum configured limit of 500000000000, set by ETH_GAS_PRICE_WEI")
```

Comment:
cool, but dont need to assert the error message

Reply:
Why?